src/protocol.h

```diff
@@ -431,12 +431,20 @@ class CInv
     std::string ToString() const;
 
     // Single-message helper methods
-    bool IsMsgTx()        const { return type == MSG_TX; }
-    bool IsMsgWtx()       const { return type == MSG_WTX; }
+    bool IsUndefined() const { return type == UNDEFINED; }
+    bool IsMsgTx() const { return type == MSG_TX; }
+    bool IsMsgBlk() const { return type == MSG_BLOCK; }
+    bool IsMsgWtx() const { return type == MSG_WTX; }
+    bool IsMsgFilteredBlk() const { return type == MSG_FILTERED_BLOCK; }
+    bool IsMsgCmpctBlk() const { return type == MSG_CMPCT_BLOCK; }
+    bool IsMsgWitnessBlk() const { return type == MSG_WITNESS_BLOCK; }
     bool IsMsgWitnessTx() const { return type == MSG_WITNESS_TX; }
+    bool IsMsgFilteredWitnessBlk() const { return type == MSG_FILTERED_WITNESS_BLOCK; }
 
     // Combined-message helper methods
-    bool IsGenTxMsg()     const { return type == MSG_TX || type == MSG_WTX || type == MSG_WITNESS_TX; }
+    bool IsGenTxMsg() const { return type == MSG_TX || type == MSG_WTX || type == MSG_WITNESS_TX; }
+    bool IsMsgBlkOrMsgWitnessBlk() const { return type == MSG_BLOCK || type == MSG_WITNESS_BLOCK; }
+    bool IsGenBlkMsg() const { return type == MSG_BLOCK || type == MSG_FILTERED_BLOCK || type == MSG_CMPCT_BLOCK || type == MSG_WITNESS_BLOCK; }
```

Comment:
Should this contain `MSG_FILTERED_WITNESS_BLOCK`?

Reply:
Well-spotted; added it in the rebase for now. The one place this helper is currently used (https://github.com/bitcoin/bitcoin/pull/19610/files#diff-eff7adeaec73a769788bb78858815c91L1763) did not include `MSG_FILTERED_WITNESS_BLOCK`, so we'd need to be sure including it doesn't change behavior (or was an oversight).

Reply:
Oh, it looks like `MSG_FILTERED_WITNESS_BLOCK` isn't actually used and is 'reserved for future use': https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki#relay. I recommend removing it from here, removing `IsMsgFilteredWitnessBlk()` (since it's not used anywhere) and adding a comment to `MSG_FILTERED_WITNESS_BLOCK` saying it's unused (I think we could even comment out the value since it's unused in the code).